catalog/spatial/wps/spatial-process-api/src/main/java/org/codice/ddf/spatial/process/api/description/LiteralDataDescription.java

```diff
@@ -0,0 +1,267 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.process.api.description;
+
+import static javax.ws.rs.core.MediaType.TEXT_PLAIN;
+
+import java.io.Serializable;
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * This class is Experimental and subject to change
+ */
+public class LiteralDataDescription implements DataDescription {
+
+    private String id;
+
+    private String name;
+
+    private String description;
+
+    private BigInteger minOccurs = BigInteger.ONE;
+
+    private BigInteger maxOccurs = BigInteger.ONE;
+
+    private Metadata metadata;
+
+    private Range range;
+
+    private Serializable defaultValue;
+
+    private Set<String> enumeratedValues = Collections.emptySet();
+
+    private DataType type;
+
+    private String unitOfMeasure;
+
+    private ValueReference reference;
+
+    //default to text/plain
+    private List<DataFormatDefinition> dataFormats = Collections.singletonList(new DataFormatDefinition(TEXT_PLAIN,
+            StandardCharsets.UTF_8.displayName()));
+
+    public LiteralDataDescription(String id, String name, String description, DataType type) {
+        this.id = id;
+        this.name = name;
+        this.description = description;
+        this.type = type;
+    }
+
+    public LiteralDataDescription(String id, String name, String description, DataType type,
+            Serializable defaultValue) {
+        this(id, name, description, type);
+        this.defaultValue = defaultValue;
+    }
+
+    public LiteralDataDescription(String id, String name, String description, Serializable defaultValue,
```

Comment:
:pencil2: Should keep the parameters in the same order for telescoping constructors (`type` should be before `defaultValue`)